### PR TITLE
CONSOLE-2400: Bump react to v17

### PR DESCRIPTION
**Pull request by zherman0**
Bump React from 16.x to 17.0.1 and other dependencies as needed.
[CONSOLE-2400](https://issues.redhat.com/browse/CONSOLE-2400)

Did not bump types/react to 17 as there were serious conflicts. Still investigating options for that.

### CODE REVIEW

**Comment by spadgett** [comment]
@zherman0 Looks like enzyme doesn't support React 17, which is causing the test errors, so we might be blocked. See https://github.com/enzymejs/enzyme/issues/2429

/cc @christianvogt

**Comment by zherman0** [comment]
/hold
We have several dependency issues that have not been resolved by third party modules.  We are holding this until 4.8 to give those external modules time to update to React 17.

**Comment by spadgett** [comment]
/override ci/prow/ceph-storage-plugin

The ceph-storage-plugin job should only have run on PRs with changes to `frontend/packages/ceph-storage-plugin`. It failed due to an unrelated CI config change. Overriding status on this PR.

**Comment by zherman0** [comment]
@spadgett - Enzyme still does not support React 17, however yarn add --dev @wojtekmaj/enzyme-adapter-react-17 is suppose to be a viable workaround.  Shall I replace the official enzyme with this personal repo and test everything?

**Comment by spadgett** [comment]
Yeah, I'm good with adding the adapter to unblock us until enzyme adds support.

File: frontend/before-tests.js
Comment (spadgett) [inline frontend/before-tests.js:4]: Remove commented-out code.

File: frontend/@types/gql/queries.d.ts
Comment (spadgett) [inline frontend/@types/gql/queries.d.ts:1]: I thought we had GraphQL types in `@types/console/generated` so this shouldn't be necessary?

cc @rawagner

Comment (rawagner) [reply frontend/@types/gql/queries.d.ts:1]: Thats right. The location changed recently in https://github.com/openshift/console/pull/8343. `@types/gql` is the old location which is not ignored by git anymore so got commited here probably.

Comment (zherman0) [reply frontend/@types/gql/queries.d.ts]: Removed the extra files

**Review by spadgett — APPROVED** [review]
/lgtm
/hold for approvals
/assign @yapei @ahardin-rh @sferich888 

@yapei This simply needs a basic regression test of console. There is no visible change to the UI.

**Pull request merged**
